daemon/libnetwork/sandbox.go

```diff
@@ -532,30 +536,29 @@ func (sb *Sandbox) EnableService() (err error) {
 			ep.enableService()
 		}
 	}
-	log.G(context.TODO()).Debugf("EnableService %s DONE", sb.containerID)
+	log.G(context.TODO()).WithField("container", sb.containerID).Debug("EnableService DONE")
 	return nil
 }
 
 // DisableService removes a managed container's endpoints from the load balancer
 // and service discovery.
-func (sb *Sandbox) DisableService() (err error) {
-	log.G(context.TODO()).Debugf("DisableService %s START", sb.containerID)
-	failedEps := []string{}
-	defer func() {
-		if len(failedEps) > 0 {
-			err = fmt.Errorf("failed to disable service on sandbox:%s, for endpoints %s", sb.ID(), strings.Join(failedEps, ","))
-		}
-	}()
+func (sb *Sandbox) DisableService() error {
+	log.G(context.TODO()).WithField("container", sb.containerID).Debug("DisableService START")
+	var failedEps []string
 	for _, ep := range sb.Endpoints() {
-		if ep.isServiceEnabled() {
-			if err := ep.deleteServiceInfoFromCluster(sb, false, "DisableService"); err != nil {
-				failedEps = append(failedEps, ep.Name())
-				log.G(context.TODO()).Warnf("failed update state for endpoint %s into cluster: %v", ep.Name(), err)
-			}
-			ep.disableService()
+		if !ep.isServiceEnabled() {
+			continue
 		}
+		if err := ep.deleteServiceInfoFromCluster(sb, false, "DisableService"); err != nil {
+			failedEps = append(failedEps, ep.Name())
+			log.G(context.TODO()).Warnf("failed update state for endpoint %s into cluster: %v", ep.Name(), err)
```

Comment:
```suggestion
			log.G(context.TODO()).WithFields(log.Fields{
				"error: err,
				"ep": ep.Name(),
			}).Warnf("failed update state for endpoint into cluster")
```

Reply:
Ah, yes; probably should add `sb.containerID` there as well. Let me update

Reply:
@akerouanton updated; PTAL